enhancements/status-crd.md

```diff
@@ -0,0 +1,250 @@
+
+#### Standard Kubernetes Conditions
+
+The status includes standard Kubernetes conditions to provide a consistent interface for monitoring tools:
+
+**Condition Types:**
+- `Ready`: True when all configuration is successfully applied to the node
+- `Degraded`: True when some configuration failed but the node is partially functional
+
+#### Failed Resources
+
+When configuration failures occur, the `failedResources` field provides detailed information about which specific resources failed and why. Each failed resource includes:
+
+- **Kind**: The type of OpenPERouter resource that failed (`Underlay`, `L2VNI`, or `L3VNI`)
+- **Name**: The name of the specific resource instance
+- **Message**: Detailed error description explaining the failure reason
+
+This structured approach allows operators to quickly identify problematic resources without parsing log files, and enables monitoring systems to create targeted alerts for specific failure types.
+
+**Condition Reasons:**
+- `ConfigurationSuccessful`: All resources configured successfully
+- `ConfigurationFailed`: Configuration failed
+
+#### Example Resources
+
+**Successful Configuration:**
+```yaml
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: RouterNodeConfigurationStatus
+metadata:
+  name: worker-1
+  namespace: openperouter-system
+  ownerReferences:
+  - apiVersion: v1
+    kind: Node
+    name: worker-1
+    uid: "12345678-1234-1234-1234-123456789abc"
+status:
+  lastUpdateTime: "2025-01-15T10:30:00Z"
+  conditions:
+  - type: Ready
+    status: "True"
+    reason: ConfigurationSuccessful
+    message: "All configuration applied successfully"
+    lastTransitionTime: "2025-01-15T10:30:00Z"
+```
+
+**Failed Configuration:**
+```yaml
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: RouterNodeConfigurationStatus
+metadata:
+  name: worker-2
+  namespace: openperouter-system
+  ownerReferences:
+  - apiVersion: v1
+    kind: Node
+    name: worker-2
+    uid: "87654321-4321-4321-4321-cba987654321"
+status:
+  lastUpdateTime: "2025-01-15T10:30:00Z"
+  failedResources:
+    - kind: Underlay
+      name: production-underlay
+      message: "Interface eth2 not present on node"
+    - kind: L2VNI
+      name: tenant-network
+      message: "VNI 100 conflicts with an existing configuration"
+  conditions:
+  - type: Ready
+    status: "False"
+    reason: ConfigurationFailed
+    message: "Configuration failed due to missing interfaces and conflicting VNI"
+    lastTransitionTime: "2025-01-15T10:30:00Z"
+  - type: Degraded
+    status: "True"
+    reason: ConfigurationFailed
+    message: "Some resources failed to configure"
+    lastTransitionTime: "2025-01-15T10:30:00Z"
+```
+
+#### Naming and Lifecycle
+
+- **Resource naming**: `<nodeName>` format (simple node name since router identity is implicit from namespace)
+- **Owner references**: RouterNodeConfigurationStatus resources are owned by their associated Node, enabling automatic cleanup when nodes are removed
```

Comment:
we should make sure that the resource is deleted also when the controller pod is removed from the node (because of a node selector / taint for example)